upf/upf_pfcp_api.c

```diff
@@ -698,6 +698,9 @@ handle_f_teid (upf_session_t * sx, upf_main_t * gtm, pfcp_pdi_t * pdi,
 
 	  if (teid)
 	    chosen = true;
+
+	  if (!(chosen) && !(create))
+	    return -1;
```

Comment:
that should IMHO be a few lines down, after the end of the `if ((pdi->f_teid.flags & F_TEID_CHID))`

A F-TEID in `Update-PDR` with F_TEID_CH = 1, F_TEID_CHID = 0 is IMHO illegal. By moving the chosen+create test after the F_TEID_CH check, a proper error for that combination is ensured.